ci/slurm/start-slurm.sh

```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-docker-compose up -d --no-build
```

Comment:
@AlexanderVanEck, this option was introduced as a way to avoid building images for each push event (it was making the CI take too long to run). Since you want to be able to use the up-to-date docker images in this PR, I recommend separating this PR into two PRs (1) for the existing `RemoteSLURMCluster` and (2) for updating the docker images. PR(2) would get merged before merging this PR. What do you think?

Reply:
I understand where you are coming from - however I think we can get away with it.

1. In the scenario where there's is no change in the Dockerfile
- We do docker-compose pull first, which pulls the latest from dockerhub
- docker-compose up --build recognises that there's no changes and uses cache for all layers

2. In the scenario where there's is a change in the Dockerfile
- We do docker-compose pull first, which pulls the latest from dockerhub
- docker-compose up --build recognises that there's some changes and uses cache for all layers that are unaffected

Drawbacks:
1. It will try to build every service in docker-compose, but it will use cache for all but the first one. f.e. https://github.com/dask/dask-jobqueue/pull/504/checks?check_run_id=3069548195#step:6:4178

Is this not what is expected?

> PR into two PRs

The "problem" here would be that the (2) PR does not test the Dockerfile before it is merged because the CI uses the old 'latest' one. and thus it is very likely that I'll push a dockerfile that won't work. 

F.e. see the failed tests in this PR which for some reason made the cluster not come up. Locally this worked.

Reply:
@andersy005 I've not yet fully understood the new CI process, do you agree with @AlexanderVanEck answer?